.github/workflows/branch-build.yml

```diff
@@ -0,0 +1,56 @@
+name: branch build
+
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Setup Node
+      uses: actions/setup-node@v1
+      with:
+        node-version: 14.x
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 2
+    - name: Cache .yarn
+      uses: c-hive/gha-yarn-cache@v1
+    - name: Cache node_modules
+      uses: actions/cache@v2
+      env:
+        cache-name: cache_node_modules
+      with:
+        path: $GITHUB_WORKSPACE/node_modules
+        key: node_modules-${{ hashFiles('**/yarn.lock')}}
+        restore-keys: |
+          node_modules-${{ hashFiles('**/yarn.lock')}}
+          node_modules-
```

Comment:
What do these two lines mean?

Reply:
Basically it tells it to look for cache keys that look like that if the initial hit didn't work. This means it will check the master branch 'cache' for a key that looks like that. In the case of the `node_modules-` it means it will get the latest cache entry. Then the yarn install will run on that and will only need to install the new dependencies or updates.